src/us/kbase/workspace/test/database/provenance/FundingReferenceTest.java

```diff
@@ -21,163 +23,229 @@
-	static final String FUNDER_ID = "ROR:04xm1d337";
-	static final String FUNDER_ID_UNTRIMMED = "\t\t\t   "  + FUNDER_ID + "  \f  \r \n\n";
-	static final String AWARD_ID = "198170392";
-	static final String AWARD_ID_UNTRIMMED = "" + AWARD_ID + "";
-	static final String AWARD_URL = "http://example.com/cgi-bin/wtf.cgi";
-	static final String AWARD_URL_UNTRIMMED = "  \t  \r\n " + AWARD_URL + "                \n";
-	static final String AWARD_TITLE = "The Best Little Toaster";
-	static final String AWARD_TITLE_UNTRIMMED = "\r\n  \f   " + AWARD_TITLE + "   \t \n\n \t";
-
+	// field names
+	static final String FUNDER_NAME = "funder name";
+	static final String FUNDER_ID = "funder ID";
+	static final String AWARD_ID = "award ID";
+	static final String AWARD_TITLE = "award title";
+	static final String AWARD_URL = "award URL";
+
+	static final String INCORRECT_FUNDER_NAME = INCORRECT + " " + FUNDER_NAME;
+	static final String INCORRECT_FUNDER_ID = INCORRECT + " " + FUNDER_ID;
+	static final String INCORRECT_AWARD_ID = INCORRECT + " " + AWARD_ID;
+	static final String INCORRECT_AWARD_TITLE = INCORRECT + " " + AWARD_TITLE;
+	static final String INCORRECT_AWARD_URL = INCORRECT + " " + AWARD_URL;
+
+	static final String FUNDER_NAME_STRING = "World's Richest Funding Body";
+	static final String FUNDER_NAME_STRING_UNTRIMMED = "\t \f \n \r" + FUNDER_NAME_STRING + "\t \f \r \n";
+	static final String FUNDER_ID_STRING = "ROR:04xm1d337";
+	static final String FUNDER_ID_STRING_UNTRIMMED = "\t\t\t   " + FUNDER_ID_STRING + "  \f  \r \n\n";
+	static final String AWARD_ID_STRING = "198170392";
+	static final String AWARD_ID_STRING_UNTRIMMED = "\n     " + AWARD_ID_STRING + "    \t";
+	static final String AWARD_URL_STRING = "http://example.com/cgi-bin/wtf.cgi";
+	static final String AWARD_URL_STRING_UNTRIMMED = "  \t  \r\n " + AWARD_URL_STRING + "                \n";
+	static final String AWARD_TITLE_STRING = "The Best Little Toaster";
+	static final String AWARD_TITLE_STRING_UNTRIMMED = "\r\n  \f   " + AWARD_TITLE_STRING + "   \t \n\n \t";
+
+	static final Map<String, String> MINIMAL_MAP = ImmutableMap.of(
+			FUNDER_NAME, FUNDER_NAME_STRING);
+
+	static final Map<String, String> ALL_MAP = ImmutableMap.of(
+			FUNDER_NAME, FUNDER_NAME_STRING,
+			FUNDER_ID, FUNDER_ID_STRING,
+			AWARD_ID, AWARD_ID_STRING,
+			AWARD_TITLE, AWARD_TITLE_STRING);
 
 	@Test
 	public void equals() throws Exception {
 		EqualsVerifier.forClass(FundingReference.class).usingGetClass().verify();
 	}
 
+	/**
+	 * Check the funding ref has the expected fields
+	 *
+	 * @param fr
+	 *                instance to check
+	 * @param expectedMap
+	 *                key/value pairs are the field names and values for the
+	 *                string fields that are expected not to be null
+	 * @param awardURL
+	 *                expected awardURL field
+	 */
+	private void assertFundingReferenceFields(FundingReference fr, final Map<String, String> expectedMap,
```

Comment:
refactor the tests so that the passing tests all use this function to check that the output object is as expected